lighthouse-core/audits/third-party-summary.js

```diff
@@ -0,0 +1,167 @@
+   * @param {string} url
+   * @return {ThirdPartyEntity|undefined}
+   */
+  static getEntitySafe(url) {
+    try {
+      return thirdPartyWeb.getEntity(url);
+    } catch (_) {
+      return undefined;
+    }
+  }
+
+
+  /**
+   *
+   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @param {Array<LH.Artifacts.TaskNode>} mainThreadTasks
+   * @param {number} cpuMultiplier
+   * @return {Map<ThirdPartyEntity, {mainThreadTime: number, transferSize: number}>}
+   */
+  static getSummaryByEntity(networkRecords, mainThreadTasks, cpuMultiplier) {
+    /** @type {Map<ThirdPartyEntity, {mainThreadTime: number, transferSize: number}>} */
+    const entities = new Map();
+
+    for (const request of networkRecords) {
+      const entity = ThirdPartySummary.getEntitySafe(request.url);
+      if (!entity) continue;
+
+      const entityStats = entities.get(entity) || {mainThreadTime: 0, transferSize: 0};
+      entityStats.transferSize += request.transferSize;
+      entities.set(entity, entityStats);
+    }
+
+    const jsURLs = BootupTime.getJavaScriptURLs(networkRecords);
+
+    for (const task of mainThreadTasks) {
+      const attributeableURL = BootupTime.getAttributableURLForTask(task, jsURLs);
+      const entity = ThirdPartySummary.getEntitySafe(attributeableURL);
+      if (!entity) continue;
+
+      const entityStats = entities.get(entity) || {mainThreadTime: 0, transferSize: 0};
+      entityStats.mainThreadTime += task.selfTime * cpuMultiplier;
+      entities.set(entity, entityStats);
+    }
+
+    return entities;
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const settings = context.settings || {};
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+    const tasks = await MainThreadTasks.request(trace, context);
+    const multiplier = settings.throttlingMethod === 'simulate' ?
+      settings.throttling.cpuSlowdownMultiplier : 1;
+
+    const summaryByEntity = ThirdPartySummary.getSummaryByEntity(networkRecords, tasks, multiplier);
+
+    const summary = {wastedBytes: 0, wastedMs: 0};
+
+    // Sort by a combined measure of bytes + main thread time.
+    // 1KB ~= 1 ms
+    /** @param {{transferSize: number, mainThreadTime: number}} stats */
+    const computeSortValue = stats => stats.transferSize / 1024 + stats.mainThreadTime;
+
+    const results = Array.from(summaryByEntity.entries())
+      .map(([entity, stats]) => {
+        summary.wastedBytes += stats.transferSize;
+        summary.wastedMs += stats.mainThreadTime;
+
+        return {
+          entity: /** @type {LH.Audit.Details.LinkValue} */ ({
+            type: 'link',
+            text: entity.name,
+            url: entity.homepage || '',
```

Comment:
at least update `Audit.Details.LinkValue`'s docs if you intend to make this a thing :)